### PR TITLE
Initial WG LTS Reformation

### LINKED ISSUE #7259: Revive WG-LTS
### Describe the issue

Microsoft recently announced that the Azure Kubernetes Service would being offering [long term support](https://aka.ms/aks/upgrades-and-lts) (LTS) for Kubernets 1.27. Some individuals at KubeConEU 2023 Contributor Summit indicated that they were either working on something similar or planned to do so, which resulted in a lively a unconference session discussing [Kubernetes LTS](https://docs.google.com/document/d/1C7utWMjGpDcBc5TB_ooddERsid59sFLC7gqLzyS5aKY/).

There appeared to be general agreement from those in attendance that "longer" support would be beneficial to the community and that some of the issues that were blockers/limitations to the previous iteration of [wg-lts](https://github.com/kubernetes/community/pull/5240) are no longer as impactful as they once were, but that additional work remains and there are still things to work through. The general consensus was also that the #1 action item from that unconference session was that we should revive WG-LTS to work on a new charter and to collaborate on the outstanding issues to try and identify a more detailed vision for what a community supported LTS would entail.

The [working group governance](https://github.com/kubernetes/community/blob/master/committee-steering/governance/wg-governance.md) covers creating and disbanding a WG, but not necessarily reviving a previously disbanded working group, so I'm opening this issue as a placeholder to discuss reviving the WG and for tracking potential items outline in the SIG/WG Lifecycle Doc (https://github.com/kubernetes/community/blob/master/sig-wg-lifecycle.md). I assume we'd just treat this as creation of a new WG, so I'll also send an email to [dev@kubernetes.io](mailto:dev@kubernetes.io) per the lifecycle doc. 

This is a topic that requires input from a number of SIGs, I've included a few of them below. 

/sig architecture release k8s-infra testing

### PR DESCRIPTION
**Pull request by jeremyrickard**
Initial PR to (re)create wg-lts. 

#7259 was opened as a tracker for this and has a lot of +1s. I also sent an email to the [dev mailing list](https://groups.google.com/a/kubernetes.io/g/dev/c/qWsMPMDqCFQ).  

This includes a draft charter that needs some eyes and reviews. 

Some things to determine:

- [ ] Identify a steering liaison
- [ ] Determine meeting times (follow up after chartering?)
- [ ] Agree on charter (part of this PR)

/committee steering

Adding stakeholder SIGs 

/sig architecture
/sig cluster-lifecycle
/sig k8s-infra
/sig release
/sig testing

SIG Approval

- [x] Architecture
- [x] Cluster Lifecycle
- [x] K8s Infra
- [x] Release
- [x] Testing
- [x] Security Response (committee)

/hold for consensus

**Which issue(s) this PR fixes**:

Fixes #7259

### CODE REVIEW

File: sigs.yaml
Comment (liggitt) [inline sigs.yaml:3259]: nit, `lts` everywhere is easier to remember (and matches all the things we used before, xref 0e9d714c7657733cebe94bb9e975cb7c121349c4)

File: wg-long-term-support/charter.md
Comment (liggitt) [inline wg-long-term-support/charter.md:47]: nit: might read easier if these are moved either to the scope or the deliverables section

**Review by liggitt — COMMENTED** [review]
a couple naming/formatting suggestions, but 👍 on the overall shape

File: sigs.yaml
Comment (xmudrii) [reply sigs.yaml:3259]: I actually wanted to leave the same comment but removed it before submitting the review. 😅 I'm not sure if `lts` is too short, but I general `lts` might be better.

File: wg-long-term-support/charter.md
Comment (jeremyrickard) [reply wg-long-term-support/charter.md:67]: thanks for catching that!

File: wg-long-term-support/charter.md
Comment (jeremyrickard) [reply wg-long-term-support/charter.md:30]: thanks for catching that!

File: sigs.yaml
Comment (jeremyrickard) [reply sigs.yaml:3259]: will do!

**Comment by jeremyrickard** [comment]
Per Chairs/TL meeting: we should probably add all SIGs as stakeholders. @BenTheElder suggested a list of "owning" SIGS and "involved" SIGS and separate things out that way.  @mrbobbytables  mentioned we should almost certainly call out node and api-machinery as stakeholders.

**Comment by jdumars** [comment]
Very glad to see this taking shape! Happy to approve when any objections/discussions are cleared.

**Comment by ameukam** [comment]
+1 with my k8s-infra SIG chair hat.

File: sigs.yaml
Comment (ritazh) [inline sigs.yaml:3253]: IMO SRC should be a stakeholder here but we can deal with that later as part of the WG.

File: wg-lts/charter.md
Comment (ritazh) [inline wg-lts/charter.md:74]:
```suggestion
This working group adheres to the Roles and Organization Management outlined in
```

File: wg-lts/charter.md
Comment (youngnick) [inline wg-lts/charter.md:81]: Should we write periodic updates and reviews to sig-arch/steering/someone else in here? Perhaps a 6-monthly update, and a review after 12 months?

Feels like having some deadlines to produce something might help keep the discussion focussed.

**Comment by aojea** [comment]
+1 as Sig-testing TL

File: wg-lts/charter.md
Comment (jeremyrickard) [reply wg-lts/charter.md:81]: @youngnick  that's a good suggestion, I will incorporate your feedback.

File: sigs.yaml
Comment (jeremyrickard) [reply sigs.yaml:3253]: I don't believe the generator/validation allows for anything other than valid SIGs here. I agree that committees could be stakeholders, but I don't think that has been a thin in the past? 

@kubernetes/steering-committee  what do you think should be done here?

Comment (jeremyrickard) [reply sigs.yaml:3253]: @ritazh I added the security response committee here as a label to capture that, for now.

**Comment by ritazh** [comment]
+1 on behalf of SRC

**Comment by youngnick** [comment]
+1 on behalf of some guy who was involved last time.

File: wg-lts/charter.md
Comment (dims) [inline wg-lts/charter.md:31]: Can you please consider some more of the items from the earlier WG charter? https://github.com/kubernetes/community/blob/master/archive/wg-lts/charter.md#in-scope

File: wg-lts/charter.md
Comment (dims) [inline wg-lts/charter.md:37]: please consider adding at least the first bullet of the previous charter's out-of-scope section. https://github.com/kubernetes/community/blob/master/archive/wg-lts/charter.md#out-of-scope

**Comment by dims** [comment]
One very critical piece this time around is there are more people in the community (especially vendors) who have public or private efforts (and people!) who help their own customers beyond community's efforts. How do we bring them to the table? (to do actual work!) Could we think of some things to get them engaged in the WG discussions from the get go and put those in the charter?

**Comment by jeremyrickard** [comment]
Thanks for the feedback @dims. I added a new commit that brings in items from the previous working group charter and added two bullet points to the in scope section to address your last comment.

**Comment by jeremyrickard** [comment]
Open question: should this PR leave the archive wg-lts for historical purposes or move the archive wg-lts out and apply the changes on top of that?

**Comment by dims** [comment]
thanks @jeremyrickard 

> Open question: should this PR leave the archive wg-lts for historical purposes or move the archive wg-lts out and apply the changes on top of that?

+1 to leave it as-is please

**Comment by pacoxu** [comment]
IMO, kubelet skew is one of the biggest problems for LTS. Have we asked SIG-Node to be one of the stakeholder SIGs?

File: wg-lts/charter.md
Comment (pacoxu) [inline wg-lts/charter.md:15]: For the survey, we have done a survey of our customers that is similar.
- The survey should cover that users may have multi clusters in multi versions. The cluster size and number of clusters also matter for upgradation. For instance, migration is another approach for multi-clusters. Some big clusters may be hard to upgrade due to complex configurations and some special dependencies(storage/networking) of pods. Some customers chose to drain old-version clusters like draining a node. `Drain old version cluster` means to migrate apps to new clusters when apps need an update.
-  How often does your cluster upgrade minor versions? How often does your cluster upgrade patch versions?

File: wg-lts/charter.md
Comment (pacoxu) [inline wg-lts/charter.md:21]: > EOL was extended by N months/years.

Extending the current long-term support of all versions?

As I know, some other software will mark some versions as long support and others as not.
- for instance, we may mark one release of a year as long support for some extended time only. This may save some effort of the community.

File: sigs.yaml
Comment (SergeyKanzhelev) [inline sigs.yaml:3256]: As @pacoxu already mentioned, let's put sig node here as well for the kubelet skew

Comment (liggitt) [reply sigs.yaml:3256]: Whether or not kubelet skew is relevant depends on the recommended changes, which are not yet known. I think ~every sig could be included as a stakeholder here, but I think it makes more sense to scope the sponsoring sigs to the ones impacted regardless of changes recommended

File: wg-lts/charter.md
Comment (liggitt) [reply wg-lts/charter.md:21]: That isn't known yet, that's one of the questions to investigate/answer

**Comment by liggitt** [comment]
> kubelet skew is one of the biggest problems for LTS. Have we asked SIG-Node to be one of the stakeholder SIGs?

That depends on what is meant by LTS and what the proposed upgrade paths would be. I would probably wait to pull in sig node until there are specific proposals that would intersect with kubelet skew

File: wg-lts/charter.md
Comment (ChaoyiHuang) [reply wg-lts/charter.md:15]: It's a little early to describe all survey questions in this initial reformation PR. Some examples are good enough.

Admit that survey questions will impact the WG-LTS conclusion greatly, need to be designed and reviewed carefully.

for exmple, if the question is expected support period for each minor version, may receive answer "the longer the better". But if the question is can you accept only one longer supported minor version every year, the answer maybe "yes".

Existing facts will be easier to answer and expressed exactly. Expectation is not equal to real need, sometimes will be overstated.

**Comment by jeremyrickard** [comment]
Thanks everyone for the reviews, comments, and discussion! 

@pacoxu raises some good points about the likely involvement of SIG Node, and really *every* SIG. I think that @liggitt is right though, that a lot will depend on what recommendations the WG actually makes and if/when we make recommendations, all appropriate SIGs will be stakeholders for those recommendations. 

I think we've had explicit approvals from the SIGs we did list as initial stakeholders, so I'm going to ping @kubernetes/steering-committee  for a review and comments now!

**Review by bryan-strassner — APPROVED** [review]

File: sigs.yaml
Comment (SergeyKanzhelev) [reply sigs.yaml:3256]: /lgtm

File: wg-lts/charter.md
Comment (bharitwal) [inline wg-lts/charter.md:95]: Would the lack of people and higher costs for providing long term support be a reason for disbanding the working group?

**Review by bharitwal — COMMENTED** [review]
LGTM. Had a few questions.

File: wg-lts/charter.md
Comment (mrbobbytables) [reply wg-lts/charter.md:15]: Agreed - probably don't need to expand too much on survey questions here. It's fine to list a few as examples, but there will likely be a significant refinement of them. We do have some data already from the regular Production Readiness Surveys.

File: wg-lts/charter.md
Comment (mrbobbytables) [reply wg-lts/charter.md:82]: Generally if the target audience is end users, docs are pushed to the primary website + blog article highlighting them.

For disruptive changes we do engage other communities more directly - e.g. pushing to reddit, CNCF end user group etc.

File: wg-lts/charter.md
Comment (mrbobbytables) [reply wg-lts/charter.md:95]: I think that is more a potential outcome of the WG. If the WG determines we currently do not have the resources, nor commitments for future resources to potentially support an LTS then theres no reason to continue.

**Review by mrbobbytables — COMMENTED** [review]
No major blockers, but a few small comments.

File: sigs.yaml
Comment (tpepper) [reply sigs.yaml:3254]: I like the idea of moving the start of the scope section here.  It reads too ambiguous as it stands.  Sure the goal is (updated?) definition for long term support in the Kubernetes project.  But for first pass reforming this needs spelled out more for the reader.

Then in the coming months if the WG bootstraps and refines the mission based on initial discussions and data gathering, that can follow in subsequent updates if not already covered by a generic/sufficient starting statement.

File: wg-lts/charter.md
Comment (tpepper) [reply wg-lts/charter.md:31]: Are there any specific ones still missing at this point you'd like included?

File: wg-lts/charter.md
Comment (palnabarun) [reply wg-lts/charter.md:46]: Resolving this since it's added.

File: wg-lts/charter.md
Comment (palnabarun) [inline wg-lts/charter.md:81]: nit:
```suggestion
- Recommend changes Kubernetes SIGs can make that will provide broad benefits in a sustainable/affordable way. This would likely take the form of one or more KEPs.
```

**Comment by palnabarun** [comment]
@jeremyrickard -- can you PTAL at the comments and also rebase the branch?

File: sigs.yaml
Comment (jeremyrickard) [reply sigs.yaml:3254]: Thanks for the suggestion, I will add this.

File: sigs.yaml
Comment (jeremyrickard) [reply sigs.yaml:3256]: @palnabarun possibly, I was holding off adding any additional SIGs given the other discussions that have occurred but I will add this one because I think it's likely to be involved regardless.

File: sigs.yaml
Comment (jeremyrickard) [reply sigs.yaml:3285]: Good point.

Comment (jeremyrickard) [reply sigs.yaml:3285]: I'll mark it as TBD for now and update later if this is approved and we have created the new group.

File: wg-lts/charter.md
Comment (jeremyrickard) [reply wg-lts/charter.md:95]: I added this as reason for disbanding for clarity.

File: wg-lts/charter.md
Comment (jeremyrickard) [reply wg-lts/charter.md:44]: Thanks!

**Comment by jeremyrickard** [comment]
@palnabarun  I don't know if we have a core set of dependencies documented. If we don't, as part of the WG we can document those and capture their support periods in order to better understand this space.  I reworded the sentence a little to indicate that.

**Comment by jeremyrickard** [comment]
Thanks to @mrbobbytables @tpepper @palnabarun @bharitwal for the most recent round of comments! 

I've rebased this PR and tried to incorporate your feedback! Please take a look again :)

Notable changes:

* Added SIG Security per @palnabarun 
* Updated mission statement in the sigs.yaml file per everyone. 
* Minor updates for other suggestions.

File: wg-lts/README.md
Comment (bharitwal) [inline wg-lts/README.md:11]:
```suggestion
The working group is organized with the goal of developing a better understanding of what "Long Term Support" might mean for Kubernetes, those who support Kubernetes, and end users. The working group will investigate changes the Kubernetes project could make related to the better understanding of what long term support might mean. The working group will also determine the feasibility, benefits, **cost**, and prerequisites  of any such changes.
```

**Review by bharitwal — APPROVED** [review]

**Comment by bharitwal** [comment]
> @jeremyrickard: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-community-verify	[48f671a](https://github.com/kubernetes/community/commit/48f671af3c0dbd4864d8186477015fdb3637b429)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/community/7287/pull-community-verify/1673431166352363520)	true	`/test pull-community-verify`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=community&pr=7287). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3Ajeremyrickard). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/community/issues?q=is:issue+is:open) when you hit one in your PR.

@jeremyrickard I see the tests are failing as `Readme.md` is an auto-generated file and sigs.yaml should have been updated.

**Comment by jeremyrickard** [comment]
> > @jeremyrickard: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > Test name	Commit	Details	Required	Rerun command
> > pull-community-verify	[48f671a](https://github.com/kubernetes/community/commit/48f671af3c0dbd4864d8186477015fdb3637b429)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/community/7287/pull-community-verify/1673431166352363520)	true	`/test pull-community-verify`
> > [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=community&pr=7287). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3Ajeremyrickard). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/community/issues?q=is:issue+is:open) when you hit one in your PR.
> 
> @jeremyrickard Apologies, I see tests are failing as `Readme.md` is an auto-generated file and sigs.yaml should have been updated.

Yeah, sorry I didn't catch that either :) I'll rebase and include your suggestion in the sigs.yaml file

**Comment by jeremyrickard** [comment]
@kubernetes/steering-committee  ping for another review, thanks!

**Review by tpepper — COMMENTED** [review]
lgtm (Steering)

File: sigs.yaml
Comment (palnabarun) [inline sigs.yaml:3258]: I volunteer to be the liaison. (unless someone else wants to be the liaison)

cc: @kubernetes/steering-committee

**Review by palnabarun — APPROVED** [review]
/lgtm
(with the steering hat on)

**Comment by palnabarun** [comment]
Steering Approvals:
- [ ] @BenTheElder 
- [ ] @cblecker  
- [x] @cpanato 
- [ ] @justaugustus 
- [x] @mrbobbytables
- [x] @palnabarun 
- [x] Tim Pepper

File: wg-lts/charter.md
Comment (MadhavJivrajani) [inline wg-lts/charter.md:73]: I see that SIG Security was added as a stakeholder, maybe an update here slipped through the cracks?

Comment (jeremyrickard) [reply wg-lts/charter.md:73]: Thanks @MadhavJivrajani, let me fix that up.

File: sigs.yaml
Comment (palnabarun) [reply sigs.yaml:3258]: SGTM either way.

File: sigs.yaml
Comment (cblecker) [inline sigs.yaml:3223]:
```suggestion
    The working group is organized with the goal of developing a better understanding
    of what "Long Term Support" might mean for Kubernetes, those who support Kubernetes,
    and end users. The working group will investigate changes the Kubernetes project
    could make related to the better understanding of what long term support might
    mean. The working group will also determine the feasibility, benefits, **cost**,
    and prerequisites of any such changes.
```

**Review by cblecker — COMMENTED** [review]
non-blocking formatting nit, otherwise lgtm.

File: wg-lts/README.md
Comment (wzshiming) [inline wg-lts/README.md:11]:
```suggestion
The working group is organized with the goal of developing a better understanding of what "Long Term Support" might mean for Kubernetes, those who support Kubernetes, and end users. The working group will investigate changes the Kubernetes project could make related to the better understanding of what long term support might mean. The working group will also determine the feasibility, benefits, **cost**, and prerequisites of any such changes.
```

File: wg-lts/charter.md
Comment (wzshiming) [inline wg-lts/charter.md:51]:
```suggestion
- Technical and end-user support: The WG may make recommendations
```

**Comment by jeremyrickard** [comment]
Looks like we now have approval from the following steering committee members:

- [ ] @BenTheElder
- [x] @cblecker
- [x] @cpanato
- [ ] @justaugustus
- [x] @mrbobbytables
- [x] @palnabarun
- [x] @tpepper

I believe this constitutes a simple majority? Are we ok to drop the hold on this and merge it? It will need another LGTM/Approve since I rolled in the suggestion from @cblecker. If possible, I would like to merge this and begin the work of spawning a mailing list and slack channels, and follow up to this with additional data as we create it.

**Comment by cblecker** [comment]
@jeremyrickard Let me check once more with Stephen and Ben. Yes, we have a majority but if they plan on reviewing we prefer to have full consensus for something as important as a new community group.

**Comment by jeremyrickard** [comment]
> @jeremyrickard Let me check once more with Stephen and Ben. Yes, we have a majority but if they plan on reviewing we prefer to have full consensus for something as important as a new community group.

Thanks @cblecker, that makes sense!

**Review by BenTheElder — COMMENTED** [review]
From a Steering Committee Perspective: LGTM. 

Sorry for the delay, this has been high on my list between vacations.

With my SIG Testing and Kubernetes maintainer hats on: I personally have reservations about the sustainability etc, but the workgroup will determine possible directions and then we can evaluate feasibility etc as a community :-)

**Comment by jdumars** [comment]
Hi Community - based on the feedback, it appears this is ready to LGTM and remove the hold, which I will do by EOD barring any objections. I echo Ben's concerns and hope that we can mobilize around this effort and make it sustainable. Thanks everyone for your support on this! 

/lgtm

**Pull request merged**
